bolt-sidecar/src/json_rpc/api.rs

```diff
@@ -113,7 +113,9 @@ impl CommitmentsRpc for JsonRpcApi {
 
         info!(?request, "received inclusion commitment request");
 
-        let tx_sender = request.tx.recover_signer().ok_or(ApiError::Custom(
+        // NOTE: request.sender is skipped from deserialization and initialized as Address::ZERO
+        // by the default Deserialization. It must be set here.
+        request.sender = request.tx.recover_signer().ok_or(ApiError::Custom(
             "failed to recover signer from transaction".to_string(),
         ))?;
```

Comment:
any better ways to do this that don't involve adding an Option<Address> in `InclusionRequest`?

Reply:
We have thought about it but ultimately this leads to create a different type with just the ec-recovered sender, which means a lot of verbosity and duplicated types. 
As such, we came to the conclusion that this little "gotcha" is better for now